### DISCUSSION TITLE
PR builds in a Turbo Monorepo

### DISCUSSION THREAD

**Discussion by gwenshap** [vercel/community#4352] [Help] [answered]
### Summary

I have a single Turbo monorepo, and I'm deploying 3 projects to Vercel from this monorepo (the plan is to have a lot more).

Every time I make a PR with a change to any workspace in the monorepo, all 3 projects are getting built and deployed. Which obviously leads to queuing. But 99% of the time, there's no reason to build and deploy all 3 when I just changed one of them.

Is there a way to tell Vercel which one of the 3 I want to build in each PR? 

### Steps to Reproduce

I'm not sure what kind of information will help here... This is the default behavior, so there's nothing special to reproduce.

I have the following workspaces in package.json:

```
  "workspaces": [
    "www", "examples/quickstart/nextjs", "examples/user_management/nextjs"],
```

**Comment by amyegan** [comment, marked as answer]
Hi @gwenshap! [Ignored Build Step](https://vercel.com/docs/projects/overview#ignored-build-step) is a useful feature for this kind of thing. You can use the `Only build Turborepo app if there are changes` option to automatically detect if the project or one of its dependencies has changed and skip the build if it is not needed. The [turbo-ignore docs](https://vercel.com/docs/monorepos/turborepo#using-turbo-ignore) have more info about how it works.
